src/ng/compile.js

```diff
@@ -2231,7 +2231,18 @@ function $CompileProvider($provide, $$sanitizeUriProvider) {
             this.$$element.removeAttr(attrName);
           } else {
             if (SIMPLE_ATTR_NAME.test(attrName)) {
-              this.$$element.attr(attrName, value);
+              // jQuery skips special boolean attrs treatment in XML nodes for
+              // historical reasons and hence AngularJS cannot freely call
+              // `.attr(attrName, false) with such attributes. To avoid issues
+              // in XHTML, call `removeAttr` in such cases instead.
+              // See https://github.com/jquery/jquery/issues/4249
+              lowercasedName = lowercase(attrName);
+              isBooleanAttr = BOOLEAN_ATTR[lowercasedName];
```

Comment:
Can we not just use `booleanKey` here rather than recomputing this stuff?

Reply:
Using `booleanKey` may be a good idea, let me try it.

Reply:
@petebacondarwin PR updated.